kolibri/plugins/management/assets/src/vue/user-page/index.vue

```diff
@@ -145,30 +144,25 @@
         // creates array of words in filter, removes empty strings
         const searchFilter = this.searchFilter.split(' ').filter(Boolean).map(
           // returns an array of search parameters, ignoring case
-          (query) => new RegExp(query, 'i'));
+          (query) => {
+            const escapedQuery = query.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&');
+            return new RegExp(escapedQuery, 'i');
+          });
 
         return this.users.filter((user) => {
           // fullname created using es6 templates
           const names = [user.full_name, user.username];
 
-          let hasRole = true;
+          let isKind = true;
           let hasName = true;
```

Comment:
Sure, and note that it's not a hard-and-fast rule. When over-used, functional style can be more confusing to read than imperative style. There are always trade-offs in terms of maintainability, brittleness, and readability...